pkg/cluster/operation/action.go

```diff
@@ -248,18 +242,11 @@ func restartInstance(getter ExecutorGetter, ins spec.Instance, timeout uint64) e
 		log.Errorf(string(stderr))
 	}
 
-	if err != nil {
+	// error or not ready
+	if err != nil || ins.Ready(e, timeout) != nil {
```

Comment:
If `err == nil` but `ins.Ready(e, timeout) != nil`, the `return errors.Annotatef(err...` will returns nil